packages/odoo/src/Adaptor.js

```diff
@@ -0,0 +1,186 @@
+import {
+  execute as commonExecute,
+  composeNextState,
+} from '@openfn/language-common';
+import { expandReferences } from '@openfn/language-common/util';
+
+import pkg from 'odoo-await';
+const Odoo = pkg;
+
+let odooConn = null;
+
+/**
+ * Execute a sequence of operations.
+ * Wraps `language-common/execute` to make working with this API easier.
+ * @example
+ * execute(
+ *   create('foo'),
+ *   delete('bar')
+ * )(state)
+ * @private
+ * @param {Operations} operations - Operations to be performed.
+ * @returns {Operation}
+ */
+export function execute(...operations) {
+  const initialState = {
+    references: [],
+    data: null,
+  };
+
+  return state => {
+    return commonExecute(login, ...operations)({ ...initialState, ...state });
+  };
+}
+
+/**
+ * Logs in to OpenSpp, gets a session token.
+ * @example
+ *  login(state)
+ * @private
+ * @param {State} state - Runtime state.
+ * @returns {State}
+ */
+async function login(state) {
+  const { baseUrl, username, password, database } = state.configuration;
+  odooConn = new Odoo({
+    baseUrl: baseUrl,
+    db: database,
+    username: username,
+    password: password,
+  });
+  try {
+    await odooConn.connect();
+  } catch (err) {
+    console.log(`✗ Error: ${err}`);
+    odooConn = null;
+  }
+  return state;
+}
+/**
+ * Create a record in Odoo
+ * @public
+ * @example
+ * create("res.partner", { name: "Kool Keith" });
+ * @function
+ * @param {string} model - The specific record model i.e. "res.partner"
+ * @param {object} data - The data to be created in JSON.
+ * @param {string} externalId - Optional external ID for the record.
+ * @returns {Operation}
+ */
+export function create(model, data, externalId) {
+  return async state => {
+    const [resolvedModel, resolvedData, resolvedExternalId] = expandReferences(
+      state,
+      model,
+      data,
+      externalId
+    );
+
+    console.log(resolvedModel, resolvedData, resolvedExternalId);
+    const response = await odooConn.create(
+      resolvedModel,
+      resolvedData,
+      resolvedExternalId
+    );
+
+    return composeNextState(state, response);
+  };
+}
+
+/**
+ * Get a record from Odoo
+ * @public
+ * @example
+ * read("res.partner", [1] , [name]);
+ * @function
+ * @param {string} model - The specific record model from i.e. "res.partner"
+ * @param {number[]} recordId - An array of record IDs to read.
+ * @param {string[]} fields - An array of fields to read from the record.
```

Comment:
Is this optional? Can we not just read all fields?

Reply:
Yes, exactly. We should include docs to clarify this (don't do it on the param,  add them to the main docs above @public, like `Returns all fields unless a field list is provided as a third argument`.

Then we need two examples: one with a caption `Download a single item with all fields` and `Download items with select fields`. In the second example, use `$.recordIds` instead of a passing a hardcoded value.